surveys/promise-rejections/survey.md

```diff
@@ -0,0 +1,211 @@
+
+## What is a Promise rejection?
+
+A Promise rejection indicates that something went wrong while executing a Promise or an async function. Rejections can occur in several situations: throwing inside an async function or a `Promise` executor/then/catch/finally callback, when calling the `reject` callback of an `executor`, or when calling `Promise.reject`.
+
+```js
+Promise.reject(new Error())  // This will result in a rejection
+
+new Promise((fulfill, reject) => {
+  reject(new Error());  // This will result in a rejection
+});
+
+new Promise(() => {
+  throw new Error();  // This will result in a rejection
+})
+
+Promise.resolve().then(() => {
+  throw new Error();  // This will result in a rejection
+})
+
+Promise.reject().then(() = {}, () => {
+  throw new Error();  // This will result in a rejection
+}
+
+Promise.reject().catch(() => {
+  throw new Error();  // This will result in a rejection
+})
+
+Promise.resolve().finally(() => {
+  throw new Error();  // This will result in a rejection
+});
+
+async function () {
+  throw new Error();  // This will result in a rejection
+}
+
+async function () {
+  if (a === undefined)
+    a();  // This will result in a rejection
+}
+```
+
+Adding the `async` keyword to a function will turn any exceptions thrown inside that function (or any throw propagated from other functions called within it) into a rejection. The same happens when refactoring callback based code that throws into async functions / Promises. Below is an example of a callback based code refactored to Promises where exceptions become rejections:
+
+## What is an unhandled rejection?
+
+There are two ways to handle rejections: by attaching a `.catch` handler to it,
+or by awaiting on the promise within a try/catch block. In both cases, the
+handling of the rejection (the execution of the callback passed to `.catch`, or
+the execution of the `catch {}` block) will happen in a future turn of the
+event loop.
+
+Promises are designed so that attaching handlers or awaiting can be done at any
+point in time, from when the Promise was created (possibly while it's still
+pending), to right before the program finishes execution.
+
+A rejection is considered unhandled from the point it happens until the point
+where a handler is attached to the Promise or the Promise is awaited within a
+`catch {}` block. Below are a few examples of handled and unhandled rejections.
```

Comment:
@benjamingr what do you think?

Reply:
There are a couple of technical inaccuracies here:

 - There are a lot more ways to handle rejections (put a promise in a Promise.all or other combinator, `resolve` another promise with it in the promise constructor etc).
 - It is not clear what "in a future turn of the event loop" means in the context of handling of the rejection but that's usually not the case - since it happens after nextTicks run but before any "event loop stuff" like timers and I/O.

But I think it's fine for us to be not 100% technically accurate when explaining terms and honestly we could keep talking about this for years :D 

I think overall the survey is LGTM and I'm +1 on sending it.